backend/api/publishing.py

```
@@ -495,13 +495,40 @@ async def _handle_pdf_storage(scenario, pdf_metadata, db):
+    Returns:
+        True if URL is a Wasabi/S3 URL, False otherwise
+    """
+    if not url or not isinstance(url, str):
+        return False
+    
+    url_lower = url.lower()
+    # Check for Wasabi endpoint patterns
+    if 'wasabisys.com' in url_lower or 'wasabi' in url_lower:
+        return True
+    # Check for AWS S3 patterns
+    if 's3.amazonaws.com' in url_lower or 's3-' in url_lower and '.amazonaws.com' in url_lower:
```

Ruff (0.14.3)

Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)